esx_service/vmdk_ops_test.py

```diff
@@ -1191,75 +1234,68 @@ def test_vmdkops_for_default_datastore(self):
                                          vm_list=[self.vm1_name])
         self.assertEqual(None, error_info)
 
-        # set access privileges on self.datastore1, and default_datastore will be set to
-        # self.datastore1 for tenant1 since this is the first privilege added for tenant1
-        # access privilege on self.datastore is mount_only
+        # After setup(), the "default_datastore" of tenant1 has been set to "__VM_DS"
+        # A full access privilege to datastore "__VM_DS" has been created for tenant1
+        # try to create a volume, which should succeed
+        # a volume will be tried to create on the default_datastore, which is self.datastore_name
+        opts =  {u'fstype': u'ext4'}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path, auth.CMD_CREATE, self.tenant1_vol1_name, opts)
+        self.assertEqual(None, error_info)
+
+        # Modify the access privilege to "__VM_DS" to disallow create
         volume_maxsize_in_MB = convert.convert_to_MB("500MB")
         volume_totalsize_in_MB = convert.convert_to_MB("2GB")
-        error_info = auth_api._tenant_access_add(name=self.tenant1_name,
-                                                 datastore=self.datastore1_name,
+        error_info = auth_api._tenant_access_set(name=self.tenant1_name,
+                                                 datastore=auth_data_const.VM_DS,
```

Comment:
Consider this scenario,

1. VM group VMG1 with __VM_DS as the default.
2. User adds two VMs VM1 and VM2 on two different datastores (DS1 and DS2) to VMG1. Each VM can create volumes on their respective datastores (DS1 and DS2).
3. Now users onVM1 and VM2 are actually isolated because they use separate datastores to keep the volumes.
4. If admin changes the privileges for __VM_DS to disallow create then users on both VMs are impacted, even though they are actually on different datastores (DS1 and DS2).

I'm not sure its correct to impact all users in every VM in the VM group in this manner.  Consider VM1 and VM2 are running two independent docker service stacks. Both are impacted if the admin removed create privileges. Admin would perhaps want to revoke privileges for a particular docker service stack on one of the VMs, but there is no way to do that here.

It may be ok as an approach but the utility of it is questionable and highly impactful. At least for VM_DS as the default, we should not allow changing the privileges as it impact unrelated users.